make.ps1

```diff
@@ -0,0 +1,235 @@
+function All-Command 
+{
+	$msBuild = FindMSBuild
+	$msBuildArguments = "/t:Rebuild /nr:false"
+	if ($msBuild -eq $null)
+	{
+		echo "Unable to locate an appropriate version of MSBuild."
+	}
+	else
+	{
+		$proc = Start-Process $msBuild $msBuildArguments -NoNewWindow -PassThru -Wait
+		if ($proc.ExitCode -ne 0)
+		{
+			echo "Build failed. If just the development tools failed to build, try installing Visual Studio. You may also still be able to run the game."
+		}
+		else
+		{
+			echo "Build succeeded."
+		}
+	}
+}
+
+function Clean-Command 
+{
+	$msBuild = FindMSBuild
+	$msBuildArguments = "/t:Clean /nr:false"
+	if ($msBuild -eq $null)
+	{
+		echo "Unable to locate an appropriate version of MSBuild."
+	}
+	else
+	{
+		$proc = Start-Process $msBuild $msBuildArguments -NoNewWindow -PassThru -Wait
+		rm *.dll
+		rm *.dll.config
+		rm mods/*/*.dll
+		rm *.pdb
+		rm mods/*/*.pdb
+		rm *.exe
+		rm ./*/bin -r
+		rm ./*/obj -r
+		echo "Clean complete."
+	}
+}
+
+function Version-Command 
+{
+	if ($command.Length -gt 1)
+	{
+		$version = $command[1]
+	}
+	elseif (Get-Command 'git' -ErrorAction SilentlyContinue)
+	{
+		$version = git name-rev --name-only --tags --no-undefined HEAD 2>$null
+		if ($version -eq $null)
+		{
+			$version = "git-" + (git rev-parse --short HEAD)
+		}
+	}
+	else
+	{	
+		echo "Unable to locate Git. The version will remain unchanged."
+	}
+	
+	if ($version -ne $null)
+	{
+		$mod = "mods/" + $modID + "/mod.yaml"
+		$replacement = (gc $mod) -Replace "Version:.*", ("Version: {0}" -f $version)
+		sc $mod $replacement
+
+		$prefix = $(gc $mod) | Where { $_.ToString().EndsWith(": User") }
+		if ($prefix -and $prefix.LastIndexOf("/") -ne -1)
+		{
+			$prefix = $prefix.Substring(0, $prefix.LastIndexOf("/"))
+		}
+		$replacement = (gc $mod) -Replace ".*: User", ("{0}/{1}: User" -f $prefix, $version)
+		sc $mod $replacement
+
+		echo ("Version strings set to '{0}'." -f $version)
+	}
+}
+
+function Test-Command
+{
+	if (Test-Path ./engine/OpenRA.Utility.exe)
+	{
+		$msg = "Testing " + $modID + " mod MiniYAML"
+		echo $msg
+		./engine/OpenRA.Utility.exe $modID --check-yaml
+	}
+	else
+	{
+		UtilityNotFound
+	}
+}
+
+function Check-Command
+{
+	if (Test-Path ./engine/OpenRA.Utility.exe)
+	{
+		echo "Checking for explicit interface violations..."
+		./engine/OpenRA.Utility.exe all --check-explicit-interfaces
+
+		$msg = "Checking for code style violations in OpenRA.Mods." + $modID + "..."
+		echo $msg
+		./engine/OpenRA.Utility.exe $modID --check-code-style OpenRA.Mods.$modID
+	}
+	else
+	{
+		UtilityNotFound
+	}
+}
+
+function Check-Scripts-Command
+{
+	if ((Get-Command "luac.exe" -ErrorAction SilentlyContinue) -ne $null)
+	{
+		echo "Testing Lua scripts..."
+		foreach ($script in ls "mods/*/maps/*/*.lua")
+		{
+			luac -p $script
+		}
+		echo "Check completed!"
+	}
+	else
+	{
+		echo "luac.exe could not be found. Please install Lua."
+	}
+}
+
+function Docs-Command
+{
+	if (Test-Path ./engine/OpenRA.Utility.exe)
+	{
+		Version-Command
```

Comment:
This will fail and give a broken error string if the template was downloaded as a zip file.